File: Back-End/src/main/java/nova/tech/com/backend/services/article/ArticleService.java
package nova.tech.com.backend.services.article;

import nova.tech.com.backend.dtos.request.ArticleRequest;
import nova.tech.com.backend.dtos.response.PageResponse;
import nova.tech.com.backend.models.Article;

public interface ArticleService {
    PageResponse getAllArticles(int page, int limit, Boolean active);
    void createArticle(ArticleRequest articleRequest);
    void updateArticle(Long id, ArticleRequest articleRequest);
    Article getArticleById(Long id);
    Article getArticleBySlug(String slug);

}

File: Front-End/src/services/article.service.ts
import { get, post, put } from "@/services/axios";
import { ArticleRequest } from "@/types/request.type";
import { Article, Page } from "@/types/response.type";

export const getAllArticles = async (page: number, limit: number, active?: boolean): Promise<Page<Article>> => {
  const res = await get<Page<Article>>(`/articles`, {
    params: active !== undefined ? { page, limit, active } : { page, limit }
  }
  );
  return res.data;
}

export const createArticle = async (articleRequest: ArticleRequest) => {
  const res = await post("/articles", articleRequest);
  return res.data;
}

export const updateArticle = async (id: number, articleRequest: ArticleRequest) => {
  const res = await put(`/articles/${id}`, articleRequest);
  return res.data;
}

export const getArticleById = async (id: number): Promise<Article> => {
  const res = await get<Article>(`/articles/${id}`);
  return res.data;
}

export const getArticleBySlug = async (slug: string): Promise<Article> => {
  const res = await get<Article>(`/articles/slug/${slug}`);
  return res.data;
}